### DISCUSSION TITLE
Button clicks stopped working after Next.js update

### DISCUSSION THREAD

**Discussion by igorgiumellizup** [cypress-io/cypress#27977] [General]
Hi.

I run e2e Cypress tests on a Web App built on Next.js.

We have over 200 test scenarios and everything was going fine and stable until the devs decided to update the Next.js version.
They went from `13.4.12` to `13.5.3`. Manually browsing, everything looks fine and no one can tell the core lib of the site was touched. 

However, the automated tests (with Cypress) started failing, specially in button clicks (running headless and normal). Clicks that worked before now don't work. Cypress clicks the button but nothing happens anymore. Even adding force:true, checking visibility before clicking, calling third party methods liked waitForStableDOM when loading the page, everything. 

Here's our Cypress version:
```
  "devDependencies": {
    "cypress": "^12.15.0",
    "cypress-split": "^1.3.15",
    "cypress-terminal-report": "^5.3.2",
    "cypress-wait-for-stable-dom": "^0.1.0"
  }
```
  
I know that this might not be related to Cypress. Any ideas on how to approach this issue? We are very worried because Next.js will be constantly updated, we can't ask the dev team to stick with a version forever.

I thought of reporting it here before going to Next.js

**Comment by manavm1990** [comment]
I am just noticing this too now with NextJS v14. I can't be sure, but it seems like this happened when I went from a server component to a `'use client'` component. That's the only change that seems to coincide with `click` not working as @igorgiumellizup describes.

I am uploading a video showing the failed test. It also shows where `click` 'happens,' but the dialog doesn't open. However, I can literally go into the Cypress rendering and click the same button and the dialog opens just fine. 😖 

https://github.com/cypress-io/cypress/assets/3387791/f75e5b5f-b4e0-4351-80e8-b67fb262df84

**Reply by manavm1990** [reply]
I am now running Next with `--turbo` flag 🎏 . Now...Cypress is working fine? As soon as I remove 🔥  that, it goes back to original problem......

**Reply by SimonyanGrno** [reply]
@igorgiumellizup @manavm1990 have you found a solution?

**Reply by igorgiumellizup** [reply]
My team dropped Nextjs so I can't confirm that anymore. @SimonyanGrno

**Comment by manavm1990** [comment]
> @igorgiumellizup @manavm1990 have you found a solution?

I'm having some other issues with Cypress upon moving to Turborepo, but regarding the initial issue I had mentioned [solution](https://github.com/cypress-io/cypress/discussions/27977?notification_referrer_id=NT_kwDOADOxj7I3OTQ1ODg1MDI1OjMzODc3OTE&notifications_query=is%3Adone#discussioncomment-7625021)
